docs/getting-started-provider-dev.md

```diff
@@ -0,0 +1,221 @@
+
+```
+export OS_PROJECT_DOMAIN_NAME=Default
+export OS_USER_DOMAIN_NAME=Default
+export OS_DOMAIN_ID=<domain_id_that_matches_name>
+export OS_PROJECT_NAME=<project_name>
+export OS_TENANT_NAME=<project_name>
+export OS_TENANT_ID=<project_id_that_matches_name>
+export OS_USERNAME=<username>
+export OS_PASSWORD=<password>
+export OS_AUTH_URL=http://<openstack_keystone_endpoint>/v3
+export OS_INTERFACE=public
+export OS_IDENTITY_API_VERSION=3
+export OS_REGION_NAME=<region_name>
+```
+
+The specific values you use will vary based on your particular environment. You may
+notice that several values are aliases of one another. This is in part because the
+values expected by the OpenStack client and
+[Gopher Cloud](https://github.com/gophercloud/gophercloud) are slightly different,
+especially with respect to the change from using `tenant` to `project`. One of our
+development goals is to make this setup easier and more consistent.
+
+### Docker
+
+Your cloud instance will need to have Docker installed. If you're ok with working from the latest
+release, it's simple enough to call the Get Docker script:
+
+```
+curl -sSL https://get.docker.io | bash
+```
+
+If you don't want to pipe a random script from the internet into your environment, you can
+install the latest version of Docker with this script.
+
+```
+sudo yum update -y
+sudo yum install -y -q epel-release yum-utils
+sudo yum-config-manager --add-repo https://download.docker.com/linux/centos/docker-ce.repo
+sudo yum-config-manager --enable docker-ce-edge
+sudo yum install -y -q docker-ce
+```
+
+However, the Kubernetes community still recommends that you run Docker v1.12. To install that version
+by hand you can use the following script.
+
+```
+sudo yum -y update
+sudo yum -y -q install yum-utils
+sudo yum-config-manager --add-repo https://yum.dockerproject.org/repo/main/centos/7
+sudo yum -y --nogpgcheck install docker-engine-1.12.6-1.el7.centos.x86_64
+```
+
+You'll want to set up Docker to use the same cgroup driver as Kubernetes
+
+```
+sed -i '/^ExecStart=\/usr\/bin\/dockerd$/ s/$/ --exec-opt native.cgroupdriver=systemd/' \
+       /usr/lib/systemd/system/docker.service
+```
+
+You may want to configure your environment to allow you to control Docker without sudo:
+
+```
+user="$(id -un 2\>/dev/null || true)"
+sudo usermod -aG docker centos
+```
+
+Regardless of how you install, enable start the service:
+
+```
+sudo systemctl daemon-reload
+sudo systemctl enable docker
+sudo systemctl start docker
+```
+
+### Development tools
+
+You're going to need a few basic development tools and applications to get, build, and run
+the source code. With your package manager you can install `git`, `gcc`, `glide`, and `etcd`.
+
+```
+sudo yum install -y -q git gcc glide etcd
+```
+
+You will also need a recent version of Go and set your environment variables.
+
+```
+GO_VERSION=1.10
+GO_ARCH=linux-amd64
+curl -o go.tgz https://dl.google.com/go/go$(GO_VERSION).$(GO_ARCH).tar.gz
```

Comment:
I think this line should be
`curl -o go.tgz https://dl.google.com/go/go${GO_VERSION}.${GO_ARCH}.tar.gz` ?